### DISCUSSION TITLE
HTML <select> control not displaying default value

### DISCUSSION THREAD

**Discussion by gramie** [vanjs-org/van#373] [General]
I have been having trouble getting a `<select>` box to display with an initial value. When I look at the example at https://vanjs.org/tutorial#state-derived-prop, I see it also does not display the correct initial value. If you change line #4 of the example:

 ` const size = van.state(16), color = van.state("black")`

to use a different colour (i.e. green):

`  const size = van.state(16), color = van.state("green")`

then the text 

> Hello 🍦VanJS

 is correctly green, but the `<select>` box displays "black". There is no way (as far as I have found) of having the `<select>` display "green" if that is the initial state of the variable "`color`".

**Comment by sirenkovladd** [comment]
The same issue discussed here https://github.com/vanjs-org/van/issues/346

Unfortunately, this is a problem in HTML itself

**Comment by Tao-VanJS** [comment]
Thanks @gramie so much for the catch!

I fixed the issue in https://github.com/vanjs-org/vanjs-org.github.io/commit/498fcfd0155f6ad7f11436c48fb45dc94839e502 by using `selected` property of the `<option>` element instead. Apparently the `value` property of the `<select>` element doesn't work, due to some oddness of HTML.

**Reply by sirenkovladd** [reply]
The problem with this approach is that when the value changes, the result of this function will be calculated N times
which can cause some performance slowdowns

**Reply by tao-corp** [reply]
Yeah. That's the tradeoff.

**Comment by gramie** [comment]
@Tao-VanJS, @sirenkovladd  Thanks very much for your help. Glad to know that it's not just me!

Another workaround I have found is to go back and set the `value` of the select to something else (e.g. "null") and then set it back to the default value. This forces any calculations to be done twice (not a problem in my application) but does have the desired effect.

`const choice = van.state(5);`
	
`	van.add($('#select'),`
`		select({ value: choice },`
`		['1', '2', '3', '4', '5'].map((x) => option({ value: x, selected: () => choice.val === x}, x)),
		)
	);`
`	choice.val = 4;`
`	choice.val = 3;`
